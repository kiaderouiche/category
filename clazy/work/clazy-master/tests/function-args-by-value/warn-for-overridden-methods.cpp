struct Trivial {};

class BaseWithVirtuals
{
public:
    virtual void virtualMethod1(const Trivial &) {}; // Warn
    virtual void virtualMethod2(const Trivial &) {}; // Warn
    void nonVirtualMethod(const Trivial &) {}; // Warn
};

class DerivedWithVirtuals : BaseWithVirtuals {
public:
    void virtualMethod1(const Trivial &) override {}; // Warn
    void virtualMethod2(const Trivial &) {}; // Warn
    void nonVirtualMethod(const Trivial &) {}; // Warn
};
