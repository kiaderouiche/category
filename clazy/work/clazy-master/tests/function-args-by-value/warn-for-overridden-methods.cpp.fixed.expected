struct Trivial {};

class BaseWithVirtuals
{
public:
    virtual void virtualMethod1(Trivial ) {}; // Warn
    virtual void virtualMethod2(Trivial ) {}; // Warn
    void nonVirtualMethod(Trivial ) {}; // Warn
};

class DerivedWithVirtuals : BaseWithVirtuals {
public:
    void virtualMethod1(Trivial ) override {}; // Warn
    void virtualMethod2(Trivial ) {}; // Warn
    void nonVirtualMethod(Trivial ) {}; // Warn
};
